src/Console/BaseCommand.php

```diff
@@ -164,6 +179,49 @@ public function initialize(): void
     {
     }
 
+    /**
+     * Returns a list of all events that will fire in the command during its lifecycle.
+     * You can override this function to add your own listener callbacks
+     *
+     * @return array<string, mixed>
+     */
+    public function implementedEvents(): array
+    {
+        return [
+            'Command.beforeExecute' => 'beforeExecute',
+            'Command.afterExecute' => 'afterExecute',
+        ];
+    }
+
+    /**
+     * Called immediately prior to the command's run method. You can use this method to configure and customize the
+     * command or perform logic that needs to happen before the command runs.
+     *
+     * @param \Cake\Event\EventInterface<\Cake\Console\BaseCommand> $event An Event instance
+     * @param \Cake\Console\Arguments $args
+     * @param \Cake\Console\ConsoleIo $io
+     * @return void
+     * @link https://book.cakephp.org/5/en/console-commands/commands.html#lifecycle-callbacks
+     */
+    public function beforeExecute(EventInterface $event, Arguments $args, ConsoleIo $io): void
+    {
+    }
+
+    /**
+     * Called immediately after the command's run method, unless an exception occurs. You can use this method to
+     * perform logic that needs to happen after the command runs.
+     *
+     * @param \Cake\Event\EventInterface<\Cake\Console\BaseCommand> $event An Event instance
+     * @param \Cake\Console\Arguments $args
+     * @param \Cake\Console\ConsoleIo $io
+     * @param int|null|void $result
```

Comment:
```suggestion
     * @param int|null $result
```

Void can't be an argument :)